src/main/java/Controller/UserAPI.java

```diff
@@ -0,0 +1,122 @@
+package Controller;
+
+import Entity.Class.User;
+import DBAccesser.User.QueryUser;
+import DBAccesser.User.InsertUser;
+import Helper.Helper;
+import Exceptions.RequiredFieldMissingException;
+import Exceptions.UserAlreadyExistsException;
+import Exceptions.InvalidLoginException;
+import Exceptions.UserNotFoundException;
+import Exceptions.APIException;
+
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.http.ResponseEntity;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+
+//this annotation tells that this class can contain methods which map to URL requests
+@RestController
+public class UserAPI {
+
+    @Autowired 
+    QueryUser queryUserHelper;
+
+    @Autowired
+    InsertUser insertUserHelper;
+
+    @Autowired
+    Helper helper;
+
+    @ExceptionHandler(UserAlreadyExistsException.class)
```

Comment:
If the unique field with which a new User is created, is Username, you can use the UsernameAlreadyExistsException class created in [#3 ](url)

Reply:
once both the codes have been merged, I will change that.